docs/blog/2022/working-with-versioned-content/index.md

```diff
@@ -0,0 +1,105 @@
+
+## Overview
+Content versioning is one of the essential features of a mature CMS. The ability to work with pages, compare and roll back when needed is a must in a collaborative environment.
+Luckily it’s also well supported by Apache Sling. In this article, I will explain how to do basic operations using only JCR and Sling APIs.
+
+<p align="center" width="100%">
+    <img class="image--with-border" src="versioned-pages.png" alt="Versioned Pages">
+    Comparing two versions of a single page on one system
+</p>
+
+## Enabling content versioning
+Sling supports versioning of JCR Resources. JCR Resources are used to represent the `content`, and as a consequence, it’s possible to version pages, assets, and tags. Unlike other Content Management Systems, we do not store scripts, configurations, and bundles in the JCR,
+for more information, see [this blog post](../configuration-application-data-in-websight-cms/).
+
+!!! Info "Before you start"
+        Make sure to replace `/content/ds-website/pages/home` in the examples below with your page path.
+ 
+
+To make a JCR Resource versionable, we need to add JCR `Mixin` to it. The easiest way to do it is to execute a script in the [Groovy Console](http://localhost:8080/apps/groovy): 
+
+```groovy
+home = session.getNode("/content/ds-website/pages/home");
+home.addMixin("mix:versionable")
+// Save is required to get apply mixin
+session.save();
+```
+
+You can verify the result with the [Resource Browser](http://localhost:8080/apps/browser#/content/ds-website/pages/home). The resource should now contain new properties.
+
+<p align="center" width="100%">
+    <img class="image--with-border" src="versioned-resource-properties.png" alt="Versioned Resource Properties">
+    New properties assigned the versioned resource
+</p>
+
+## Creating multiple page versions:
+Again, let's use Groovy script to create a couple of versions of a homepage:
+```groovy
+home = session.getNode("/content/ds-website/pages/home");
+
+for (int i = 0; i< 10; i++) {
+  home.checkout();
+  home.checkin();
+}
+
+println home.getVersionHistory();
+```
+
+The code above should create 10 versions of a page and print the complete history once done. Note that version creation is relatively expensive, so we should not overuse it.
+Each version of a resource is stored as a `jcr:fronzenNode` under `/jcr:system/jcr:versionStorage/<generated hash>`.
+
+In my case, the path is `/jcr:system/jcr:versionStorage/5f/2b/64/5f2b64d2-fdb9-4ed5-af8a-cd92a6971f66`.
+
+## Accessing page versions:
+There are multiple ways we can access the page versions.
+
+- Directly by accessing frozen node with .html extension.
+
+In my case, it's `http://localhost:8080/jcr:system/jcr:versionStorage/5f/2b/64/5f2b64d2-fdb9-4ed5-af8a-cd92a6971f66/1.1002/jcr:frozenNode/jcr:content.html`, but it will be unique on every system.
+
+- Programmatically by using JCR API [1]
+
+- Using Sling API for getting available versions and fetching versioned content [2]
```

Comment:
Fixed formatting:
```suggestion
There are multiple ways we can access the page versions:

- Directly by accessing frozen node with .html extension.
    * In my case, it's `http://localhost:8080/jcr:system/jcr:versionStorage/5f/2b/64/5f2b64d2-fdb9-4ed5-af8a-cd92a6971f66/1.1002/jcr:frozenNode/jcr:content.html`, but it will be unique on every system.
- Programmatically by using JCR API [1]
- Using Sling API for getting available versions and fetching versioned content [2]
```